src/block/Lectern.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\tile\Lectern as TileLectern;
+use pocketmine\block\utils\BlockDataSerializer;
+use pocketmine\block\utils\FacesOppositePlacingPlayerTrait;
+use pocketmine\block\utils\HorizontalFacingTrait;
+use pocketmine\entity\Location;
+use pocketmine\entity\object\ItemEntity;
+use pocketmine\item\Item;
+use pocketmine\item\ItemFactory;
+use pocketmine\item\WritableBookBase;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+
+class Lectern extends Transparent {
+	use FacesOppositePlacingPlayerTrait;
+	use HorizontalFacingTrait;
+
+	protected int $page = 0;
+	protected int $totalPages = 0;
+	protected ?Item $book = null;
+
+	public function readStateFromData(int $id, int $stateMeta) : void{
+		$this->facing = BlockDataSerializer::readLegacyHorizontalFacing($stateMeta & 0x03);
+	}
+
+	protected function writeStateToMeta() : int{
+		return BlockDataSerializer::writeLegacyHorizontalFacing($this->facing);
+	}
+
+	public function getStateBitmask() : int{
+		return 0b11;
+	}
+
+	public function readStateFromWorld() : void{
+		parent::readStateFromWorld();
+		$tile = $this->position->getWorld()->getTile($this->position);
+
+		if($tile instanceof TileLectern) {
+
+			$this->book = $tile->getBook();
+			$this->page = $tile->getPage();
+			$this->totalPages = $tile->getTotalPages();
+
+		}
+	}
+
+	public function writeStateToWorld() : void{
+		parent::writeStateToWorld();
+		$tile = $this->position->getWorld()->getTile($this->position);
+		if($tile instanceof TileLectern){
+			$tile->setBook($this->book);
+			$tile->setPage($this->page);
+			$tile->setTotalPages($this->totalPages);
+		}
+	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+
+		if($item instanceof WritableBookBase) {
+			$this->book = $item->pop();
+			$this->position->getWorld()->setBlock($this->position, $this);
+			return true;
+		}
+		return false;
+	}
+
+	public function onAttack(Item $item, int $face, ?Player $player = null) : bool{
+		$tile = $this->position->getWorld()->getTile($this->position);
+		if($this->book !== null && !$this->book->isNull() && $tile instanceof TileLectern)
+		{
+			$droppedBook = new ItemEntity(Location::fromObject($this->position->up(), $this->position->getWorld(), 0, 0), $this->book);
+			$droppedBook->spawnToAll();
+
+			$tile->setBook(ItemFactory::air());
+			$tile->setPage(0);
+			$tile->setTotalPages(0);
+			$this->readStateFromWorld();
```

Comment:
This should not be here. Any interactions with tile data should be done in `writeStateToWorld()` only. Update the block's properties and then use `setBlock()` to update the world.